https://github.com/kowainik/extensions/pull/81
From 81cb8768fb4e1e58b7aa4fb57561f0df791436b4 Mon Sep 17 00:00:00 2001
From: hololeap <hololeap@users.noreply.github.com>
Date: Thu, 22 Sep 2022 12:23:21 -0600
Subject: [PATCH] Add flag to toggle building executable

Some distributors need to use the old Cabal-v1 build systems, which
will build the 'extensions' executable even if it isn't needed (for
instance when this package is only needed as a dependency for stan).

This adds a new 'executable' flag (enabled by default) which will allow
for disabling the 'extensions' executable if it isn't needed.

Signed-off-by: hololeap <hololeap@users.noreply.github.com>
---
 extensions.cabal | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/extensions.cabal b/extensions.cabal
index db233bb..d645932 100644
--- a/extensions.cabal
+++ b/extensions.cabal
@@ -19,6 +19,11 @@ tested-with:         GHC == 8.8.4
                      GHC == 8.10.7
                      GHC == 9.0.2
 
+flag executable
+  description: Build the extensions executable
+  default: True
+  manual: True
+
 source-repository head
   type:                git
   location:            https://github.com/kowainik/extensions.git
@@ -79,6 +84,8 @@ library
 
 executable extensions
   import:              common-options
+  if !flag(executable)
+    buildable: False
   hs-source-dirs:      app
   main-is:             Main.hs
   other-modules:       Cli
